stl/inc/mutex

```diff
@@ -34,24 +34,22 @@ _EXPORT_STD class condition_variable_any;
 struct _Mtx_internal_imp_mirror {
 #ifdef _CRT_WINDOWS
 #ifdef _WIN64
-    static constexpr size_t _Critical_section_size = 8;
+    static constexpr size_t _Critical_section_size = 16;
 #else // _WIN64
-    static constexpr size_t _Critical_section_size = 4;
+    static constexpr size_t _Critical_section_size = 8;
 #endif // _WIN64
 #else // _CRT_WINDOWS
 #ifdef _WIN64
-    static constexpr size_t _Critical_section_size = 56;
+    static constexpr size_t _Critical_section_size = 64;
 #else // _WIN64
-    static constexpr size_t _Critical_section_size = 32;
+    static constexpr size_t _Critical_section_size = 36;
 #endif // _WIN64
 #endif // _CRT_WINDOWS
 
+    static constexpr size_t _Critical_section_align = alignof(void*);
+
     int _Type;
-    const void* _Vptr;

```

Comment:
The intent of `_Vptr` was to allow constant initialization. Ideally, it should be initialized with the address of the virtual function table of `stl_critical_section_win7`.

I've implemented the strategy for static linking, but haven't found a proper way for dynamic linking. See #2285.

Reply:
Oh, in that case it does need to know the layout.

I intend to get rid of the vtable. We know the dynamic type of this member, so we can just use static dispatch. Until then, we should ensure that the vtable is not unexpectedly used.

Reply:
I agree that the vtable can eventually be eliminated and that there is no reason to mention it here. We need only to preserve the size, alignment, and the location of the member we directly observe.

Reply:
Hmm... I'm afraid that old binaries still need to read the vtable.

Reply:
Do they? AFAIK, the entire interface from the headers to the DLL/LIB goes through a flat C API.